com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs

```diff
@@ -934,6 +938,22 @@ public void OnAfterDeserialize()
                 k_AssetVersion = 8;
             }
 
+            if (k_AssetVersion < 9)
+            {
+                if (m_AdditionalLightsShadowResolutionTierHigh == AdditionalLightsDefaultShadowResolutionTierHigh &&
```

Comment:
Maybe add a function "isCustomShadowResolution" or a local boolean at least.

Reply:
Thanks for the suggestion, I added an intermediate variable in https://github.com/Unity-Technologies/Graphics/commit/82f939df2d54238754f8cb91df530258f5bf9212